parser/v2/cssparser_test.go

```diff
@@ -38,8 +41,9 @@ func TestExpressionCSSPropertyParser(t *testing.T) {
 			},
 		},
 		{
-			name:  "css: single constant property with windows newlines",
-			input: "background-color:\r\n{ constants.BackgroundColor };\r\n",
+			name:        "css: single constant property with windows newlines",
+			input:       "background-color:\r\n{ constants.BackgroundColor };\r\n",
+			expectedCSS: "background-color: '                           ';\n",
```

Comment:
> The expected CSS is presumably the output after parsing, which is essentially the same as what we get if we've formatted the code.

I believe this to be the behavior that is causing all of these length mismatches. Since the entire document is parsed into a series of nodes and then built back up with the formatting specified in all of the `Write()` methods, this is the default output after the changes I've made with the ContextWriter.

Reply:
I added the suggested line length checking to all of the unit tests that I updated. There were quite a few mismatches outside of the ones you caught so those are failing as of https://github.com/a-h/templ/pull/636/commits/4c7959bea9f4c5c15ee16d975b477043187f711a.

Reply:
I've start on _a_ solution in another branch on my fork: https://github.com/alehechka/templ/compare/feature/html-lsp...alehechka:templ:fix/mismatch-line-lengths.

The rough idea is to include a `perservePositions` bool on the ContextWriter that will determine how certain nodes get formatted. As of this comment, this is resolving the manipulation of self-closing tags. The next step is reconciling extraneous white space between the source and output.